middleware/http/server_test.go

```diff
@@ -30,80 +30,105 @@ func TestHTTPHandlerWrapping(t *testing.T) {
 	var (
 		spanRecorder = &recorder.ReporterRecorder{}
 		tr, _        = zipkin.NewTracer(spanRecorder, zipkin.WithLocalEndpoint(lep))
-		httpRecorder = httptest.NewRecorder()
-		requestBuf   = bytes.NewBufferString("incoming data")
-		responseBuf  = bytes.NewBufferString("oh oh we have a 404")
 		headers      = make(http.Header)
 		spanName     = "wrapper_test"
 		code         = 404
+		request      *http.Request
 	)
+
 	headers.Add("some-key", "some-value")
 	headers.Add("other-key", "other-value")
 
-	request, err := http.NewRequest("POST", "/test", requestBuf)
-	if err != nil {
-		t.Fatalf("unable to create request")
-	}
+	testCases := []struct {
+		method         string
+		body           *bytes.Buffer
+		hasRequestSize bool
+	}{
```

Comment:
⬇️ there are the three possible cases for this feature:
1. `POST` with body
2. `POST` with empty body
3. `GET` with no body